error-reporting-cpp/src/error_message_internal.h

```diff
@@ -0,0 +1,220 @@
+
+#ifndef ERROR_REPORTING_CPP_ERROR_MESSAGE_INTERNAL_H
+#define ERROR_REPORTING_CPP_ERROR_MESSAGE_INTERNAL_H
+
+#include <regex>
+#include <sstream>
+#include <utility>
+#include <string_view>
+
+/** @file
+ * @brief Internal header file for Exasol error-reporting-cpp
+ *
+ * Includes classes and macros used to store the internal linked list of error codes.
+ *
+ */
+
+
+/**
+ * @brief Global error reporting namespace.
+ *
+ * Used to encapsulate error reporting classes.
+ */
+namespace error_reporting {
+
+    /**
+     * @brief Internal struct, representing a node of the internal linked list of error code messages.
+     *
+     * Contains all information for a particular error code, which are required to build the error collection JSON.
+     * Also contains a pointer to the next element of the linked list.
+     */
+    struct error_message_declaration_internal {
+        explicit error_message_declaration_internal(
+                const std::string_view & _error_code,
+                const std::string_view & _error_message,
+                const std::string_view & _error_mitigations,
+                size_t _nr_parameters,
+                const char* const* _parameters);
+
+        std::string str();
+        const std::string_view & error_code;
+        const std::string_view & error_message;
+        const std::string_view & error_mitigations;
+        const size_t nr_parameters;
+        const char * const * parameters;
+        error_message_declaration_internal * next;
+    };
+
+    /**
+     * @brief Global error reporting container.
+     *
+     * Contains the root node of the internal error code linked list, which is used to build the error collection JSON.
+     * Defines a method to build the JSON from the internal linked list, \see error_message_declaration_container::collect_error_messages_as_json()
+     */
+    class error_message_declaration_container {
+    public:
+        /**
+         * @brief Constructor.
+         */
+        error_message_declaration_container() : root(nullptr) {}
+
+        /**
+         * @brief Adds a new node to the internal linked list of error-codes.
+         */
+        void register_error_message(error_message_declaration_internal * msg) {
+            if (nullptr == root) {
+                root = msg;
+            } else {
+                msg->next = root;
+                root = msg;
+            }
+        }
+
+        /**
+         * @brief Collects all elements in the list as JSON object.
+         *
+         * @return A string containing all registered error-codes in the following format:
+         *          @code{.json}
+         *          [{ "code": "EC_1" ,  "message": "Error Message 1",
```

Comment:
- First question, is this the format the crawler expects.
- Second, for the error code, we need to replace underscore with dash.